### ISSUE TITLE
AliasesList::unalias_mime_type() uses slow linear search in a sorted Vec

### ISSUE THREAD

**Issue by nyanpasu64** [ebassi/xdg-mime-rs#26]
I'm writing a program that heavily utilizes MIME alias resolution. I've noticed that xdg-mime's alias resolution function (`AliasesList::unalias_mime_type()`) takes linear runtime in the number of MIME aliases on the current system, which scales poorly (my system has around 300).

https://github.com/ebassi/xdg-mime-rs/blob/23204a3e112447e5d8fdf7b2301f5e9d6f29fe13/src/alias.rs#L68-L73

Oddly enough, it appears that `AliasesList.aliases` is sorted every time you call `add_aliases()` (so hopefully it's never out of order), yet `unalias_mime_type()` calls `.iter().find()` which performs a linear search. I think it would probably be faster, and scale better, to turn the vector into a slice, then run binary search ([binary_search](https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search) or [binary_search_by](https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search_by)).

Note that I haven't done any benchmarks yet. I probably should run benchmarks, and afterwards either submit a PR to fix this, or add a comment saying that binary search isn't faster in practice.

**Comment by ebassi** [comment]
Yes, a microbenchmark would be much appreciated. The alias data set isn't huge, so it might not make any difference in practice.

**Comment by Cheong-Lau** [comment]
I just did a quick benchmark, and I found that on 2000 iterations of `unalias_mime_type`, using `binary_search_by` was ~32.5% faster, which is definitely faster.

Unfortunately the tests failed when I did that, no real idea why 🤷‍♂️ . Maybe there's more than one match?
```
unalias_mime_type       time:   [1.1403 ms 1.1405 ms 1.1408 ms]
                        change: [-32.758% -32.449% -32.177%] (p = 0.00 < 0.05)
                        Performance has improved.
```
